tools/kms_secrets_encryption.py

```diff
@@ -0,0 +1,548 @@
+        
+        return file_path_decrypted
+
+
+    def decrypt_tfvars_secrets(self):
+        """A method that decrypts the secrets contained in the terraform.tfvars file.
+
+        This method contains the logic for handling the decryption of the secrets 
+        and any file paths associated with it using GCP KMS. Once decrypted, it calls 
+        write_new_tfvars() to write all secrets to a new terraform.tfvars file. 
+        """    
+
+        # Abort the decryption if the tfvars is in plaintext using kms_cryptokey_id as an indicator
+        if not self.tfvars_data.get("kms_cryptokey_id"):
+            print("No kms_cryptokey_id present in tfvars. Ensure the secrets are encrypted and try again.\n")
+            raise SystemExit()
+
+        # Abort the decryption if GCP credentials is missing
+        if not self.tfvars_data.get("gcp_credentials_file"):
+            print("Missing gcp_credentials_file in tfvars. Ensure gcp_credentials_file is valid and try again.\n")
+            raise SystemExit()
+
+        # Set crypto key path to use kms_cryptokey_id
+        self.crypto_key_path  = self.tfvars_data.get("kms_cryptokey_id")
+
+        # Decrypt all secrets
+        try:
+            for secret in self.tfvars_secrets:
+                # Additional handling needed if the string is a path to a file (IE. cam_credentials_file)
+                if os.path.isfile(self.tfvars_secrets.get(secret)):
+                    file_decrypted = self.decrypt_file(self.tfvars_secrets.get(secret))
+                    self.tfvars_secrets[secret] = file_decrypted
+                else:
+                    print("Decrypting {}...".format(secret))
+                    plaintext = self.decrypt_ciphertext(self.tfvars_secrets.get(secret))
+                    self.tfvars_secrets[secret] = plaintext
+            
+            # Write encrypted secrets into new terraform.tfvars file
+            self.write_new_tfvars()
+            print("\nSuccessfully decrypted all secrets!\n")
+
+        except Exception as err:
+            print("An exception occurred decrypting secrets:")
+            print("{}\n".format(err))
+            raise SystemExit()
+
+
+    def encrypt_file(self, file_path):
+        """A method that encrypts the contents of a text file.
+
+        Uses GCP KMS to encrypt the plaintext in a file to ciphertext using 
+        the provided symmetric crypto key that belongs to this instance.
+        
+        Args:
+            file_path (str): the path of the text file being encrypted
+        Returns:
+            string: the path to the encrypted text file created
+        """
+
+        try:
+            print("Encrypting file: {}...".format(file_path))
+            
+            with open(file_path) as f:
+                f = json.load(f)
+        
+            f_string = json.dumps(f)
+            f_encrypted_string = self.encrypt_plaintext(f_string)
+            file_path_encrypted = "{}.encrypted".format(file_path)
```

Comment:
Here is what I think you are doing. On encrypting a file, say "a.txt", you encrypt it and save it as "a.txt.encrypted", then rename "a.txt" to "a.txt.backup".  On decrypting, you expect the file to end in ".encrypted", say "b.txt.encrypted", you decrypt it and take out the .encrypted to get "b.txt", then you delete "b.txt.encrypted".
how about:
on encryption: take a.txt, encrypt it and call it a.txt.encrypted. leave a.txt alone
on decryption: take b.txt, decrypt it and call it b.txt.decrypted. leave b.txt alone